src/lib/autocomplete/autocomplete-core.ts

```diff
@@ -399,22 +412,42 @@ export class AutocompleteCore extends ListDropdownAwareCore implements IAutocomp
   }
 
   private _onFilterComplete(): void {
-    if (!this._adapter.hasFocus()) {
+    const shouldSelectOnBlur = this._selectFirstOptionOnBlur && this._selectOnBlurPending;
+
+    if (!this._options.length) {
+      this._selectOnBlurPending = false;
       if (this._isDropdownOpen) {
         this._closeDropdown();
       }
+      if (shouldSelectOnBlur && !this._allowUnmatched) {
+        this._filterText = '';
+        this._adapter.setSelectedText('');
+      }
```

Comment:
I didn't catch this originally because it didn't support updating when a value was already selected, but looks like we might need to additionally call `_clearValue()` or something similar.  I notice that if a value is selected, and I type a string with no matches and tab off, the filter text clears, but `forge-autocomplete-change` never emits and the "Selected Value" in the demo app isn't cleared.  Aside from that, though, all the updates look good and I'm excited about this feature!